crates/pallet-feeds/src/lib.rs

```diff
@@ -354,19 +360,16 @@ impl<T: Config> Call<T> {
                 };
                 let feed_processor = T::feed_processor(feed_processor_id);
                 let objects_mappings = feed_processor.object_mappings(*feed_id, object);
-
                 objects_mappings
                     .into_iter()
-                    .map(|object_mapping| {
-                        // Scope the key of the object to the feed_id namespace
-                        let key = crypto::sha256_hash_pair(feed_id.encode(), object_mapping.key);
-                        FeedObjectMapping {
-                            key,
-                            // `FeedId` is the first field in the extrinsic. `1+` corresponds to `Call::put {}`
-                            // enum variant encoding.
-                            // update the offset to include the absolute offset in the extrinsic
-                            offset: 1 + mem::size_of::<T::FeedId>() as u32 + object_mapping.offset,
-                        }
+                    .filter_map(|object_mapping| {
+                        let mut co = object_mapping.try_into_call_object(
+                            feed_id,
+                            object.as_slice(),
+                            |data| crypto::sha256_hash(data),
+                        )?;
+                        co.offset += 1 + mem::size_of::<T::FeedId>() as u32;
```

Comment:
I would strongly prefer to have a comment here explaining these offsets for yourself, reviewers and prosperity just like it was done before. Since it is a fixed offset for all mappings, might move into a variable even.

Reply:
true. Missed this in the second refactor 🤷🏼